### PR TITLE
feat: migrate ChatJS to AI SDK 7

### PR DESCRIPTION
**Pull request by FranciscoMoretti**
Upgrade ChatJS and generated applications from AI SDK 6 to the coherent `ai 7.0.93` / `@ai-sdk/react 4.0.96` provider-v4 release set. This is a standalone change against main; main still matches the research baseline `8422c767`. Refs #307.

Generation calls now use the v7 instructions, lifecycle, telemetry, usage, file and tool contracts. Full generated history uses `responseMessages`. Gateway models no longer rely on v3 assertions, and the incompatible Evalite model wrapper is removed. Explicit OpenTelemetry integrations preserve opt-in tracing. Manifests, peers, Node >=22 requirements, Bun lockfile, changesets and generated-template compatibility are updated together.

Thread preserves canonical identity, metadata, restored tool state and prior errors across reconnects. Full replays replace content, while continuations seed saved parts into SDK 7's fresh response state. All five mapped resume regressions pass with their original assertions. The database schema, persisted data and demo configuration are unchanged.

Validation on Node 24.20.0 / Bun 1.3.11:

- `bun install --frozen-lockfile`, `bun lint`, `bun test:types`: pass.
- `bun test:unit`: 13 worktree checks, 34 CLI tests, 65 Thread tests and 149 chat tests pass. Includes existing branch/parallel/cancellation/approval ownership and transport/file/usage cases, plus focused replay, restored-tool, FileData, legacy usage render and message-boundary tests.
- `bun test:e2e`: 4 Chromium page smoke tests pass on the default preset. These are page smoke, not live provider evidence.
- `bun template:sync` and `bun template:check`: pass. Exclude local devtools recordings from generated templates.
- Live browser on the existing direct OpenAI credentials: gpt-5-mini reasoning/text completion; wordCount execution and custom renderer; retry into a sibling and switching back; reload during generation through the real Redis-backed resume endpoint; concurrent gpt-5-mini/gpt-5-nano responses with selected-mini cancellation while nano continues to completion. Persisted siblings remain selectable. Next MCP reports no compilation/runtime errors on those journeys.

Limitations:

- The configured Vercel gateway credential returns Unauthorized, which was exercised through the app error UI. Successful live tests used temporary local OpenAI defaults and a refreshed OpenAI catalog; both were restored exactly before committing. Default-gateway live success still needs a working credential.
- The app has no approve/deny action, so live approval UI coverage is unavailable; controlled Thread approval ownership tests pass. Live MCP/OAuth, deep research, media generation and Langfuse export are not claimed verified.
- Package-specific npm peer overrides cover only the tested Files SDK storage and Evalite task/scorer boundaries; a fresh npm 11 scaffold install passes; the unused AI adapter is not validated. Evalite task/scorers remain usable without its incompatible SDK-6 tracing wrapper. Supported deprecated `result.toUIMessageStream` and the legacy OTel span adapter remain narrow documented adaptations.

Details and reproducible validation notes: [docs/ai-sdk-7-migration.md](docs/ai-sdk-7-migration.md). No merge, deployment or package publication is requested by this PR.

## Summary by Sourcery

Migrate ChatJS and its generated applications to AI SDK 7 while preserving thread history, reconnect behavior, telemetry, and backward compatibility with existing persisted data.

New Features:
- Upgrade ChatJS and generated applications to AI SDK 7 and provider-v4 APIs.
- Preserve assistant identity, metadata, restored tool state, and prior errors across thread reconnects.

Bug Fixes:
- Prevent duplicated or lost message content during full replay and continuation resumes.
- Maintain compatibility with persisted SDK 6 usage metadata while displaying SDK 7 token details.
- Handle SDK 7 file data formats and dynamic tool persistence correctly.

Enhancements:
- Migrate generation lifecycle, response history, usage, provider options, and telemetry contracts to SDK 7.
- Remove incompatible Evalite SDK 6 model wrapping while retaining supported opt-in OpenTelemetry integrations.
- Update gateway model interfaces and provider implementations to use provider-v4 types without legacy assertions.

Build:
- Pin AI SDK 7/provider-v4 dependencies, update peer requirements and overrides, and require Node.js 22 or newer.
- Refresh the Bun lockfile and generated-template synchronization to match the new dependency set.

Documentation:
- Add migration guidance covering compatibility boundaries, reconnect behavior, and validation results.
- Update Thread usage documentation for AI SDK 7 dependencies and reconnect architecture.

Tests:
- Add coverage for SDK 7 resume replay, restored tools, legacy and nested usage metadata, file data handling, and persistence boundaries.

---
## Summary by cubic
Upgrades ChatJS and generated applications from AI SDK 6 to the provider-v4 `ai 7.0.93` / `@ai-sdk/react 4.0.96` release set. Generation calls use the v7 `instructions`, `responseMessages`, telemetry, and lifecycle contracts, and Thread reconnects preserve canonical identity, metadata, restored tool state, and prior errors. This is a standalone change against main; main still matches the research baseline `8422c767`. Refs #307.

**Compatibility**

- Requires Node >=22 and peer `ai@^7.0.93` / `@ai-sdk/react@^4.0.96`; manifests, lockfile, changesets, and generated templates are updated together.
- Removes Evalite's SDK-6 model wrapper; sample evals lose automatic tracing until Evalite supports provider v4.
- Keeps Files SDK's optional SDK-6 peer for storage/file APIs only; its AI adapter is not validated.
- Preserves opt-in tracing via `@ai-sdk/otel`'s legacy adapter; `result.toUIMessageStream` remains supported.
- Reads persisted usage metadata with SDK 6 legacy field fallbacks while leaving the database schema and data unchanged; unsupported reasoning-file/custom parts are rejected rather than persisted.
- Details and validation notes are in [docs/ai-sdk-7-migration.md](docs/ai-sdk-7-migration.md).

<sup>Written for commit ae718bddf731b4b1abf345c7d5b0fbe17f210e8e. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/FranciscoMoretti/chat-js/pull/308?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Summary by CodeRabbit

- **New Features**
  - Added compatibility with AI SDK 7 for chat responses, tools, and model providers.
  - Improved stream reconnection to preserve assistant identity, metadata, tool state, and errors.
  - Added telemetry coverage for chat generation and research workflows.
  - Updated curated model availability.

- **Bug Fixes**
  - Corrected reasoning and cached-token usage reporting, including legacy conversations.
  - Improved handling of file URLs and inline file data.
  - Normalized invalid MCP tool descriptions safely.

- **Documentation**
  - Added AI SDK 7 migration and reconnection guidance.

### CODE REVIEW

File: apps/chat/components/ai-elements/context.tsx [hunk 1]
```diff
@@ -316,7 +316,7 @@ export const ContextReasoningUsage = ({
 	...props
 }: ContextReasoningUsageProps) => {
 	const { usage, modelId } = useContextValue();
-	const reasoningTokens = usage?.reasoningTokens ?? 0;
```
[the remark this answers is not in the document]
Comment (FranciscoMoretti) [reply apps/chat/components/ai-elements/context.tsx:319]: [from Codex]: Fixed in 126dcf2b. Both usage displays now use a shared reader that prefers SDK 7 nested values and falls back to persisted SDK 6 fields, including preserving zero values. A render regression test reproduces the original TypeError and passes with the fix; types, lint and the full unit suite pass.

File: apps/chat/tools/platform/deep-research/types.ts [hunk 2]
```diff
@@ -28,9 +29,17 @@ export function createTelemetry(
   options: Pick<AgentOptions, "messageId" | "requestId">
 ) {
   return {
-    isEnabled: true,
-    functionId,
-    metadata: {
+    telemetry: {
+      integrations: chatTelemetry,
+      isEnabled: true,
+      functionId,
```
[the remark this answers is not in the document]
Comment (FranciscoMoretti) [reply apps/chat/tools/platform/deep-research/types.ts:35]: [from Codex]: Checked against baseline 8422c767 and the installed SDK 6 TelemetrySettings: both input and output recording were already enabled by default, and both cited paths already opted into telemetry. This migration deliberately preserves existing Langfuse behavior using LegacyOpenTelemetry. Disabling content recording would be a separate observability policy change, not a migration regression fix, so I am retaining the existing behavior.

File: packages/thread/package.json [hunk 3]
```diff
@@ -54,8 +54,8 @@
 		"test:types": "tsc -p tsconfig.json --noEmit && bun run build"
 	},
 	"peerDependencies": {
-		"@ai-sdk/react": "^3.0.246",
-		"ai": "^6.0.244",
+		"@ai-sdk/react": "^4.0.96",
+		"ai": "^7.0.93",
```
[the remark this answers is not in the document]
Comment (FranciscoMoretti) [reply packages/thread/package.json:58]: [from Codex]: Fixed in 126dcf2b with package-scoped npm overrides for the optional AI peers of Evalite and Files SDK. The incompatible Evalite model wrapper has been removed; evaluation tasks/scorers and the tested Files storage boundary remain. A freshly generated scaffold now completes npm 11 install without force or legacy-peer-deps. The migration document records these compatibility boundaries.

File: .changeset/chatjs-sdk-seven.md [hunk 4]
```diff
@@ -0,0 +1,6 @@
+---
+"@chat-js/thread": minor
```
[the remark this answers is not in the document]
Comment (FranciscoMoretti) [reply .changeset/chatjs-sdk-seven.md:2]: [from Codex]: These packages are pre-1.0: thread is 0.1.0 and CLI is 0.8.0. The minor changesets therefore produce 0.2.0 and 0.9.0; ^0.1.0 does not accept 0.2.0. This signals the breaking change at the pre-1.0 minor boundary without prematurely declaring thread 1.0. The changeset explicitly states the new AI SDK/React peers and Node >=22 requirement. The removed Evalite wrapper belongs to the app, not the published thread API. Retaining the minor bumps intentionally.

File: apps/chat/package.json [hunk 5]
```diff
@@ -159,7 +160,11 @@
   "overrides": {
     "commander": "^14.0.0",
     "evalite": {
-      "better-sqlite3": "$better-sqlite3"
+      "better-sqlite3": "$better-sqlite3",
+      "ai": "$ai"
```
[the remark this answers is not in the document]
Comment (FranciscoMoretti) [reply apps/chat/package.json:164]: [from Codex]: These overrides intentionally target npm in generated standalone apps, not the Bun workspace. Removing them restores the reproduced npm 11 ERESOLVE failure. The CLI supports four package managers; current CI now passes all eight install combinations (with and without Electron). The remaining SDK 6 optional peers in bun.lock are documented compatibility boundaries for unused adapters, not the active app model graph. A global AI override would unnecessarily broaden the change. Retaining the scoped npm fix.
